src/main/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -271,7 +263,12 @@ public boolean hasNext() {
 
     @Override
     public SAMRecord next() {
-        return iterator.next();
+        SAMRecord samRecord = iterator.next();
+        if (validationStringency != ValidationStringency.SILENT) {
+            SAMUtils.processValidationErrors(samRecord.isValid(), samRecordIndex++, validationStringency);
```

Comment:
Since `samRecordIndex` is a class not a local variable, can you add a comment to the declaration saying that it is only used when Validation Stringency is not SILENT, and otherwise it isn't valid. (this was always true even before this PR, but it would be good to make that clear).